core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -442,6 +442,8 @@ open class TileInfo {
      * @see <a href="https://github.com/yairm210/Unciv/wiki/uniques#user-content-tilefilter">tileFilter</a>
      */
     fun matchesFilter(filter: String, civInfo: CivilizationInfo? = null): Boolean {
+        if (filter == "unimproved Forest")
```

Comment:
That settles it... we're going to have to create sub-uniques so we can have "unimproved []" 
Hardcoding both the filter and the Forest value is too anti for me.